src/actions/actions.ts

```diff
@@ -2685,8 +2687,26 @@ class ActionChangeInVisualBlockMode extends BaseCommand {
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
     const deleteOperator = new DeleteOperator();
 
+    // Check to see number in lines of block
+    let blockLines = 0;
+    let blockStart = position;
+    let blockEnd = position;
+    for (const {start, end} of Position.IterateLine(vimState)) {
+      blockStart = start;
+      blockEnd = end;
+      blockLines += 1;
+    }
+
+    // Do a regular visual 'change' if only 1 line of block is selected
```

Comment:
Will remove the special case and improve the insert block mode a little bit more, visual block is just a little funky, I will try to fix the root funkiness there instead of this way